File: java/folder program/Smartphone.java
public class Smartphone extends Gadget {
    private String sistemOperasi;
    private String memori;
    private String kamera;

    public Smartphone() {}

    public Smartphone(String kode,String nama, int harga, String deskripsi,
                      String jenis, String garansi, String status, String stok,
                      String sistemOperasi, String memori, String kamera) {
        super(kode,nama, harga, deskripsi, jenis, garansi, status, stok);
        this.sistemOperasi = sistemOperasi;
        this.memori = memori;
        this.kamera = kamera;
    }

    // Getter & Setter
    public String getSistemOperasi() { return sistemOperasi; }
    public void setSistemOperasi(String sistemOperasi) { this.sistemOperasi = sistemOperasi; }

    public String getMemori() { return memori; }
    public void setMemori(String memori) { this.memori = memori; }

    public String getKamera() { return kamera; }
    public void setKamera(String kamera) { this.kamera = kamera; }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Sistem Operasi: " + sistemOperasi);
        System.out.println("Memori: " + memori);
        System.out.println("Kamera: " + kamera);
    }
}

File: cpp/folder program/Smartphone.cpp
#include <iostream>
#include "Gadget.cpp"   // mewarisi dari Gadget

using namespace std;

class Smartphone : public Gadget {
private:
    string sistemOperasi;
    string memori;
    string kamera;

public:
    Smartphone() {
    }

    // Setter & Getter
    void setSistemOperasi(string sistemOperasi) {
        this->sistemOperasi = sistemOperasi;
    }
    string getSistemOperasi() {
        return this->sistemOperasi;
    }

    void setMemori(string memori) {
        this->memori = memori;
    }
    string getMemori() {
        return this->memori;
    }

    void setKamera(string kamera) {
        this->kamera = kamera;
    }
    string getKamera() {
        return this->kamera;
    }
};
